src/Npgsql/NpgsqlConnector.cs

```diff
@@ -446,72 +449,102 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
 
             try
             {
-                await RawOpen(timeout, async, cancellationToken);
+                var potentialHosts = Settings.Host!.Split(',');
+                var hostIndex = 0;
+                while(ConnectedHost == null && hostIndex < potentialHosts.Length)
```

Comment:
What if we move the host checking to the `NpgsqlConnection.Open` and the pool? That way, we can cache the host on the pool (so every new open connection will just connect to the same host).

Reply:
Sure, I can take a look at that. I think that's actually necessary to achieve something I want to in the "next version" anyway:

> I'd like to not have to kill off connections and actually pool them in a different bucket (i.e. it's not unreasonable to have an application with the same connectionstring except for TargetServerType being primary in some cases and Secondary in others; I'd like to be able to move connections between the Pools as the server moves around without disconnecting/reconnecting. I started implementing a PoolKey concept to start looking at that, however it's not strictly required for the current implementation and I've removed it again)

Reply:
Please, only 'take a look', as I would like to also hear @roji's opinion on this (and I would feel really bad if you will spend to much time on something we haven't decided yet :rofl:).

Reply:
I'll leave this till last and probably whack any changes into another branch off this one anyway so I've got this implementation for reference/safety